pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -274,6 +280,10 @@ func (a *AddVolumeOptions) Validate(isAddOp bool) error {
 				if len(a.SecretName) == 0 {
 					return errors.New("must provide --secret-name for --type=secret")
 				}
+			case "configmap":
+				if len(a.ConfigMapName) == 0 {
+					return errors.New("must provide --configmap-name for --type=configmap")
```

Comment:
@fabianofranz do we not have a better mechanism to ensure that this relationship is enforced?